JSON encode error when indexer SearchKey Filter is not empty

### SearchKey

```
s := &indexer.SearchKey{
		Script:     lockScript,
		ScriptType: types.ScriptTypeLock,
		Filter: &indexer.Filter{
			Script: &types.Script{
				CodeHash: types.HexToHash("0x89cd8003a0eaf8e65e0c31525b7d1d5c1becefd2ea75bb4cff87810ae37764d8"),
				HashType: types.HashTypeType,
				Args:     lockScript.Hash().Bytes()[:20],
			},
		},
	}
```

### Error message
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6c6634]

goroutine 1 [running]:
encoding/json.(*encodeState).marshal.func1()
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:327 +0x6e
panic({0x721f00, 0x9ade60})
        /home/dylan/Documents/release/go/src/runtime/panic.go:884 +0x212
github.com/nervosnetwork/ckb-sdk-go/v2/indexer.Filter.MarshalJSON.func1(...)
        /home/dylan/go/pkg/mod/github.com/nervosnetwork/ckb-sdk-go/v2@v2.1.0/indexer/json.go:20
github.com/nervosnetwork/ckb-sdk-go/v2/indexer.Filter.MarshalJSON({0xc00009e870, 0x0, 0x0, 0x0, 0x0})
        /home/dylan/go/pkg/mod/github.com/nervosnetwork/ckb-sdk-go/v2@v2.1.0/indexer/json.go:26 +0x34
encoding/json.marshalerEncoder(0xc0000ce300, {0x71a120?, 0xc0000a9998?, 0x405f09?}, {0x1a?, 0x3b?})
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:478 +0xbe
encoding/json.structEncoder.encode({{{0xc0000e0480?, 0x411a84?, 0x7f62047a7228?}, 0xc0000a9b30?}}, 0xc0000ce300, {0x751340?, 0xc0000a9980?, 0xc000017390?}, {0x0, 0x1})
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:760 +0x1f4
encoding/json.ptrEncoder.encode({0xc000014098?}, 0xc0000ce300, {0x7012a0?, 0xc0000a9980?, 0x7012a0?}, {0x8?, 0xe1?})
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:944 +0x25e
encoding/json.(*encodeState).reflectValue(0x717520?, {0x7012a0?, 0xc0000a9980?, 0xc0000c97c8?}, {0xb2?, 0x2a?})
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:359 +0x78
encoding/json.interfaceEncoder(0xc0000ce300, {0x717520?, 0xc0000a9a10?, 0x405f09?}, {0xa0?, 0xed?})
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:715 +0xc8
encoding/json.arrayEncoder.encode({0xc000017250?}, 0xc0000ce300, {0x708300?, 0xc000012360?, 0x463b91?}, {0xf4?, 0x62?})
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:915 +0xd5
encoding/json.sliceEncoder.encode({0xc000014078?}, 0xc0000ce300, {0x708300?, 0xc000012360?, 0x708300?}, {0x0?, 0x0?})
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:888 +0x367
encoding/json.(*encodeState).reflectValue(0x0?, {0x708300?, 0xc000012360?, 0x4106c7?}, {0x78?, 0x0?})
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:359 +0x78
encoding/json.(*encodeState).marshal(0x18?, {0x708300?, 0xc000012360?}, {0x0?, 0x0?})
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:331 +0xfa
encoding/json.Marshal({0x708300, 0xc000012360})
        /home/dylan/Documents/release/go/src/encoding/json/encode.go:160 +0x45
github.com/ethereum/go-ethereum/rpc.(*Client).newMessage(0x0?, {0x77b1f6, 0x9}, {0xc0000a9a10?, 0x3, 0x3})
        /home/dylan/go/pkg/mod/github.com/ethereum/go-ethereum@v1.10.26/rpc/client.go:472 +0x11d
github.com/ethereum/go-ethereum/rpc.(*Client).CallContext(0xc0000ce280, {0x808c10, 0xc00001e0d0}, {0x701260, 0xc0000a99e0}, {0x77b1f6, 0x9}, {0xc0000a9a10, 0x3, 0x3})
        /home/dylan/go/pkg/mod/github.com/ethereum/go-ethereum@v1.10.26/rpc/client.go:297 +0x11d
github.com/nervosnetwork/ckb-sdk-go/v2/indexer.(*client).GetCells(0xc000014070, {0x808c10, 0xc00001e0d0}, 0xc0000a9980, {0x779d55, 0x3}, 0x1?, {0x0, 0x0})
        /home/dylan/go/pkg/mod/github.com/nervosnetwork/ckb-sdk-go/v2@v2.1.0/indexer/indexer.go:59 +0x18f
```

--- a/indexer/indexer_test.go
+++ b/indexer/indexer_test.go
@@ -2,6 +2,8 @@ package indexer
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	ethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/nervosnetwork/ckb-sdk-go/v2/types"
 	"github.com/stretchr/testify/assert"
@@ -96,6 +98,30 @@ func TestGetTransactionsGrouped(t *testing.T) {
 	assert.NotEmpty(t, resp.Objects[0].Cells[0])
 	assert.NotEmpty(t, resp.Objects[0].Cells[0].IoType)
 	assert.NotEmpty(t, resp.Objects[0].Cells[0].IoIndex)
+}
+
+func TestFilter(t *testing.T) {
+	lockScript := &types.Script{
+		CodeHash: types.HexToHash("0x58c5f491aba6d61678b7cf7edf4910b1f5e00ec0cde2f42e0abb4fd9aff25a63"),
+		HashType: types.HashTypeType,
+		Args:     ethcommon.FromHex("0xe53f35ccf63bb37a3bb0ac3b7f89808077a78eae"),
+	}
+	filter := &Filter{
+		Script: &types.Script{
+			CodeHash: types.HexToHash("0x58c5f491aba6d61678b7cf7edf4910b1f5e00ec0cde2f42e0abb4fd9aff25a63"),
+			HashType: types.HashTypeType,
+			Args:     lockScript.Hash().Bytes()[:20],
+		},
+	}
+	s := &SearchKey{
+		Script:     lockScript,
+		ScriptType: types.ScriptTypeLock,
+		Filter:     filter,
+	}
+	jsonF, _ := json.Marshal(filter)
+	fmt.Println(string(jsonF))
+	_, err := c.GetTransactions(context.Background(), s, SearchOrderAsc, 1000, "")
+	checkError(t, err)
 }
 
 func checkError(t *testing.T, err error) {

--- a/indexer/json.go
+++ b/indexer/json.go
@@ -7,19 +7,23 @@ import (
 )
 
 type jsonCellsFilter struct {
-	Script              *types.Script     `json:"script"`
-	ScriptLenRange      [2]hexutil.Uint64 `json:"script_len_range,omitempty"`
-	OutputDataLenRange  [2]hexutil.Uint64 `json:"output_data_len_range,omitempty"`
-	OutputCapacityRange [2]hexutil.Uint64 `json:"output_capacity_range,omitempty"`
-	BlockRange          [2]hexutil.Uint64 `json:"block_range,omitempty"`
+	Script              *types.Script      `json:"script"`
+	ScriptLenRange      *[2]hexutil.Uint64 `json:"script_len_range,omitempty"`
+	OutputDataLenRange  *[2]hexutil.Uint64 `json:"output_data_len_range,omitempty"`
+	OutputCapacityRange *[2]hexutil.Uint64 `json:"output_capacity_range,omitempty"`
+	BlockRange          *[2]hexutil.Uint64 `json:"block_range,omitempty"`
 }
 
 func (r Filter) MarshalJSON() ([]byte, error) {
-	toUint64Array := func(a *[2]uint64) [2]hexutil.Uint64 {
-		result := [2]hexutil.Uint64{}
-		result[0] = hexutil.Uint64(a[0])
-		result[0] = hexutil.Uint64(a[1])
-		return result
+	toUint64Array := func(a *[2]uint64) *[2]hexutil.Uint64 {
+		if a == nil {
+			return nil
+		} else {
+			result := [2]hexutil.Uint64{}
+			result[0] = hexutil.Uint64(a[0])
+			result[1] = hexutil.Uint64(a[1])
+			return &result
+		}
 	}
 	jsonObj := &jsonCellsFilter{
 		Script:              r.Script,

